nixos/modules/services/databases/influxdb2.nix

```diff
@@ -53,6 +110,57 @@ in
         KillMode = "control-group";
         Restart = "on-failure";
       };
+
+      # Mark if this is the first startup so postStart can do the initial setup
+      preStart = mkIf cfg.provision.enable ''
+        if ! test -e "$STATE_DIRECTORY/influxd.bolt"; then
+          touch "$STATE_DIRECTORY/.first_startup"
+        fi
+      '';
+
+      postStart = let
+        initCfg = cfg.provision.initialSetup;
+      in mkIf cfg.provision.enable (
+        ''
+          set -euo pipefail
+          export INFLUX_HOST="http://"${escapeShellArg (cfg.settings.http-bind-address or "localhost:8086")}
+
+          # Wait for the influxdb server to come online
+          count=0
+          while ! ${getExe pkgs.influxdb2-cli} ping &>/dev/null; do
+            if [ "$count" -eq 300 ]; then
+              echo "Tried for 30 seconds, giving up..."
+              exit 1
+            fi
+
+            if ! kill -0 "$MAINPID"; then
+              echo "Main server died, giving up..."
+              exit 1
+            fi
+
+            sleep 0.1
+            count=$((count++))
+          done
+
+          # Do the initial database setup. Pass /dev/null as configs-path to
+          # avoid saving the token as the active config.
+          if test -e "$STATE_DIRECTORY/.first_startup"; then
+            ${getExe pkgs.influxdb2-cli} setup \
+              --configs-path /dev/null \
+              --org ${escapeShellArg initCfg.organization} \
+              --bucket ${escapeShellArg initCfg.bucket} \
+              --username ${escapeShellArg initCfg.username} \
+              --password "$(< ${escapeShellArg initCfg.passwordFile})" \
+              --token "$(< ${escapeShellArg initCfg.tokenFile})" \
```

Comment:
What would happen when `tokenFile` is `null`

Reply:
Good point, either we can make it non optional or skip specifying it when it is null. The problem with the latter is that full provisioning requires this token to be known, so maybe making it mandatory is the right call. What are your thoughts?

Reply:
We can default it to `$STATE_DIRECTORY/token` and generate it if non-exist. But for the sake of simplicity and the declarative spirit of nixos, let's just make it mandatory until someone complains.

Reply:
Also we can (and should IMO) make use of `LoadCredential`.

Reply:
Done